app/src/main/java/com/sapuseven/untis/activities/MainActivity.kt

```diff
@@ -575,12 +575,16 @@ class MainActivity :
 
 			lines.add(startTimeInt)
 			lines.add(endTimeInt)
+			labels.add(hour.label)
```

Comment:
Why not check for empty labels right away?
```suggestion
			labels.add(if (hour.label.isBlank()) (index + 1).toString() else hour.label)
```